File: DSA/Ayush_Mishra_590014844/Day_51_Question_2.py
def smallest_subarray_sum_gt_x(arr, x):
    left = 0
    sum_ = 0
    min_len = len(arr) + 1

    for right in range(len(arr)):
        sum_ += arr[right]
        while sum_ > x:
            min_len = min(min_len, right - left + 1)
            sum_ -= arr[left]
            left += 1
    return min_len if min_len <= len(arr) else 0

arr = list(map(int, input().split()))
x = int(input())
print(smallest_subarray_sum_gt_x(arr, x))

File: DSA/Ayush_Mishra_590014844/Day_51_Question_2.c
#include <stdio.h>

int smallestSubarray(int arr[], int n, int x) {
    int minLen = n + 1, sum = 0;
    int left = 0;

    for (int right = 0; right < n; right++) {
        sum += arr[right];
        while (sum > x) {
            if (right - left + 1 < minLen)
                minLen = right - left + 1;
            sum -= arr[left];
            left++;
        }
    }
    if (minLen == n + 1)
        return 0;
    return minLen;
}

int main() {
    int n, x, arr[1000];
    scanf("%d", &n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    scanf("%d", &x);
    printf("%d\n", smallestSubarray(arr, n, x));
    return 0;
}
